dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -3550,6 +3550,35 @@ ReturnedColumn* buildReturnedColumn(Item* item, gp_walk_info& gwi, bool& nonSupp
   return rc;
 }
 
+// parse the boolean fields to string "true" or "false"
+ReturnedColumn* buildBooleanConstantColumn(Item* item, gp_walk_info& gwi, bool& nonSupport)
+{
+  ConstantColumn* cc = NULL;
+
+  if (gwi.thd)
+  {
+    {
+      if (!item->fixed())
+      {
+        item->fix_fields(gwi.thd, (Item**)&item);
+      }
+    }
+  }
+  cc = new ConstantColumnSInt(colType_MysqlToIDB(item), (int64_t)item->val_int() ? "true" : "false",
+                              (int64_t)item->val_int());
+
+  if (cc)
+    cc->timeZone(gwi.timeZone);
+
+  if (cc && item->name.length)
+    cc->alias(item->name.str);
+
+  if (cc)
+    cc->charsetNumber(item->collation.collation->number);
```

Comment:
Are you sure that collation will never be null in the context? The same is actual about item->name.

Reply:
Before defining the function `buildBooleanConstColumn`,All the arguments will enter the function `rc = buildReturnedColumn(ifp->arguments()[i], gwi, nonSupport)` [here](https://github.com/qggcs/mariadb-columnstore-engine/blob/f4aba866ecf4d6e4c2aa3fc6e52808741e79b790/dbcon/mysql/ha_mcs_execplan.cpp#L4049).
after debugging, the `item.collation` and `item.name` have default value, So it will not be null. these code in `buildBooleanConstColumn` is same as [buildReturnedColumn](https://github.com/qggcs/mariadb-columnstore-engine/blob/f4aba866ecf4d6e4c2aa3fc6e52808741e79b790/dbcon/mysql/ha_mcs_execplan.cpp#L3544-L3548).